File: Codeground/FrogJump/frog_jump.cpp
#define _CRT_SECURE_NO_WARNINGS
#include <iostream>

using namespace std;

int Answer;
int a[1000000];

int main(int argc, char** argv)
{
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(0);
  int T, test_case;
  // freopen("frog_jump_input.txt", "r", stdin);
  cin >> T;
  for (test_case = 0; test_case < T; test_case++)
  {
    int N, K;

    cin >> N;   
    for (int i = 0; i < N; i++) {
      cin >> a[i];
    }
    cin >> K;

    int before_jump = 0;
    int before_stone = 0;
    int jump_num = 0;
    bool can_jump = true;

    for (int i = 0; i < N; i++) {

      if (a[i] - before_stone > K) {
        can_jump = false;
        break;
      }

      if (a[i] - before_jump > K) {
        jump_num++;
        before_jump = before_stone;
      }
      before_stone = a[i];

    }

    if (can_jump)
      Answer = jump_num + 1;
    else
      Answer = -1;

    cout << "Case #" << test_case + 1 << endl;
    cout << Answer << endl;
  }

  return 0;
}

File: Codeground/FrogJump/frog_jump.py
import sys
import os
import csv


class Unbuffered(object):
    def __init__(self, stream):
        self.stream = stream

    def write(self, data):
        self.stream.write(data)
        self.stream.flush()

    def writelines(self, datas):
        self.stream.writelines(datas)
        self.stream.flush()

    def __getattr__(self, attr):
        return getattr(self.stream, attr)


if os.name == 'nt':
    inf = open('test.txt', 'r')
    # inf = open('frog_jump_input.txt', 'r')
else:
    inf = sys.stdin
sys.stdout = Unbuffered(sys.stdout)

line = []
line_len = 0
line_index = 0


def readstr():
    global line
    global line_len
    global line_index
    if line_index >= line_len:
        line = inf.readline().split()
        line_index = 0
        line_len = len(line)
    line_index += 1
    return line[line_index - 1]


def readint():
    return int(readstr())


T = readint()
for t in range(T):
    N = readint()
    a = []
    for i in range(N):
        a.append(readint())
    K = readint()

    before_jump = 0
    before_stone = 0
    jump_num = 0
    can_jump = True

    for i in range(N):
        if a[i] - before_stone > K:
            can_jump = False
            break
        if a[i] - before_jump > K:
            jump_num += 1
            before_jump = before_stone
        before_stone = a[i]


    if can_jump:
        Answer = jump_num + 1
    else:
        Answer = -1

    print(f'Case #{t+1}')
    print(Answer)

inf.close()
